cmd/acra-translator/acra-translator.go

```diff
@@ -162,6 +172,56 @@ func main() {
 		os.Exit(0)
 	})
 
+	go sigHandlerSIGHUP.RegisterWithContext(mainContext)
+	sigHandlerSIGHUP.AddCallback(func() {
+		log.Infof("Received incoming SIGHUP signal")
+		log.Debugf("Stop accepting new connections, waiting until current connections close")
+		readerServer.StopListeners()
+
+		// Set env flag for forked process
+		if err := os.Setenv(GracefulEnv, "true"); err != nil {
+			log.WithError(err).Errorln("Unexpected error on os.Setenv, graceful restart won't work. Please check env variables, especially gracefulEnv")
+		}
+
+		var fdHTTP, fdGRPC uintptr
+		if listener := readerServer.GetHTTPListener(); listener != nil {
+			fdHTTP, err = network.ListenerFileDescriptor(readerServer.GetHTTPListener())
```

Comment:
You can use `listener` variable inside the if clause instead of getting it again.

Reply:
Agree, I'd suggest to use existing `listener`.

Reply:
That's true. Thank you